src/NodaTime.Test/DateIntervalTest.cs

```diff
@@ -296,7 +296,42 @@ public void Intersection(string firstInterval, string secondInterval, string exp
             var expectedResult = ParseInterval(expectedInterval);
             Assert.AreEqual(expectedResult, value.Intersection(other));
         }
-        
+
+        [Test]
+        public void Union_NullInterval_Throws()
+        {
+            var value = new DateInterval(new LocalDate(100), new LocalDate(200));
+            Assert.Throws<ArgumentNullException>(() => value.Union(null));
+        }
+
+        [Test]
+        public void Union_DifferentCalendar_Throws()
+        {
+            var value = new DateInterval(
+                new LocalDate(2017, 11, 6, CalendarSystem.Gregorian),
+                new LocalDate(2017, 11, 10, CalendarSystem.Gregorian));
+
+            var other = new DateInterval(
+                new LocalDate(2017, 11, 6, CalendarSystem.Coptic),
+                new LocalDate(2017, 11, 10, CalendarSystem.Coptic));
+
+            Assert.Throws<ArgumentException>(() => value.Union(other));
+        }
+
+        [TestCase("2014-03-07,2014-03-10", "2014-03-12,2014-03-17", null)]
```

Comment:
I think you can use Description="..." to give descriptions to each test case. That would definitely help here :)